File: Module/ALittleIDE/Other/GameLibrary/Core/Script/Core/Utility/Math.lua

module("ALittle", package.seeall)

local ___pairs = pairs
local ___ipairs = ipairs


local tonumber = tonumber
local floor = math.floor
local random = math.random
local randomseed = math.randomseed
local abs = math.abs
local cos = math.cos
local sin = math.sin
function Math_Sin(v)
	return sin(v)
end

function Math_Cos(v)
	return cos(v)
end

function Math_Abs(v)
	return abs(v)
end

function Math_Floor(v)
	return floor(v)
end

function Math_ToInt(s)
	return floor(tonumber(s))
end

function Math_ToDouble(s)
	return tonumber(s)
end

function Math_RandomSeed(seed)
	randomseed(seed)
end

function Math_RandomInt(min, max)
	return random(min, max)
end

function Math_RandomDouble(min, max)
	return random() * (max - min) + min
end

File: ALittle/Module/ALittleIDE/Other/GameLibrary/Core/JSScript/Core/Utility/Math.js
{
if (typeof ALittle === "undefined") window.ALittle = {};


ALittle.Math_Min = function(x, y) {
	if (x < y) {
		return x;
	}
	return y;
}

ALittle.Math_Max = function(x, y) {
	if (x < y) {
		return y;
	}
	return x;
}

ALittle.Math_Sin = function(v) {
	return Math.sin(v);
}

ALittle.Math_Tan = function(v) {
	return Math.tan(v);
}

ALittle.Math_Cos = function(v) {
	return Math.cos(v);
}

ALittle.Math_Abs = function(v) {
	return Math.abs(v);
}

ALittle.Math_Floor = function(v) {
	return Math.floor(v);
}

ALittle.Math_Ceil = function(v) {
	return Math.ceil(v);
}

ALittle.Math_Sqrt = function(v) {
	return Math.sqrt(v);
}

ALittle.Math_ToInt = function(s) {
	let value = parseInt(s);
	if (isNaN(value)) {
		return undefined;
	} else {
		return value;
	}
}

ALittle.Math_ToIntOrZero = function(s) {
	let value = ALittle.Math_ToInt(s);
	if (value === undefined) {
		return 0;
	}
	return value;
}

ALittle.Math_ToDouble = function(s) {
	let value = parseFloat(s);
	if (isNaN(value)) {
		return undefined;
	} else {
		return value;
	}
}

ALittle.Math_ToDoubleOrZero = function(s) {
	let value = ALittle.Math_ToDouble(s);
	if (value === undefined) {
		return 0;
	}
	return value;
}

ALittle.Math_RandomSeed = function(seed) {
}

ALittle.Math_RandomInt = function(min, max) {
	return Math.floor(Math.random() * (max - min + 1)) + min;
}

ALittle.Math_RandomDouble = function(min, max) {
	return Math.random() * (max - min) + min;
}

}